test/java/seedu/addressbook/commands/AddCommandTest.java

```diff
@@ -0,0 +1,159 @@
+package seedu.addressbook.commands;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Test;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+public class AddCommandTest {
+    private static final List<ReadOnlyPerson> LAST_SHOWN_LIST = Collections.emptyList();
+    
+    @Test
+    public void addCommand_invalidName_throws() {
+        Set<String> tags = new HashSet<>();
+        final String[] invalidNames = { "", " ", "[]\\[;]" };
+        for (String name : invalidNames) {
+            assertConstructingInvalidAddCmdThrows(name, Phone.EXAMPLE, false, Email.EXAMPLE, false,
+                    Address.EXAMPLE, false, tags);
+        }
+    }
+    
+    @Test
+    public void addCommand_invalidPhone_throws() {
+        Set<String> tags = new HashSet<>();
+        final String[] invalidNumbers = { "", " ", "1234-5678", "[]\\[;]", "abc", "a123", "+651234" };
+        for (String number : invalidNumbers) {
+            assertConstructingInvalidAddCmdThrows(Name.EXAMPLE, number, false, Email.EXAMPLE, false,
+                    Address.EXAMPLE, false, tags);
+        }
+    }
+    
+    @Test
+    public void addCommand_invalidEmail_throws() {
+        Set<String> tags = new HashSet<>();
+        final String[] invalidEmails = { "", " ", "def.com", "@", "@def", "@def.com", "abc@" };
```

Comment:
OK, these test cases adequately test the invalid inputs where the right side of the `@` is empty and the left side of the `@` is empty.

In addition, maybe test for characters that we reject on the right side and left side of the `@`?